lib/mantav2-migrate/migrator.js

```diff
@@ -0,0 +1,746 @@
+                                    webapi.isV2 = false;
+                                    webapi.err = new VError(
+                                        err,
+                                        'could not "GET %s/" to check webapi instance %s (%s)',
+                                        webapi.adminIp,
+                                        webapi.uuid,
+                                        webapi.alias
+                                    );
+                                    nextWebapi();
+                                });
+
+                                req.end();
+                            }
+                        },
+                        function finish(err) {
+                            next(err);
+                        }
+                    );
+                }
+            ]
+        },
+        function finish(err) {
+            cb(err, wInfo);
+        }
+    );
+};
+
+/*
+ * If necessary, read in
+ *      /var/db/snaplink-cleanup/discovery/{region}_{shard}_sherlock.tsv.gz
+ * files for each shard and write out "delink" scripts to be run on
+ * (a) each storage node and, then (b) on each shard's moray to "delink"
+ * each snaplink.
+ *
+ * "Delinking" a snaplink is:
+ *  - generate a new object id (uuid) for the delinked object,
+ *  - [on storage node] create a hardlink from the old object id file to the new
+ *    object id (fast, takes no extra space so we need not worry about disk
+ *    space on storage nodes for this), and
+ *  - [moray] update the metadata entry to have the new object id
+ *
+ * Generate files are:
+ *      /var/db/snaplink-cleanup/delink/
+ *          {region}_{storage_id}_stordelink.sh
+ *          {region}_{shard}_moraydelink.sh
+ *
+ * This calls back `function (err, scripts, generated)` where, on success,
+ * `scripts` is:
+ *      {
+ *          stordelink: [
+ *              '{region}_{storage_id}_stordelink.sh',
+ *              ...
+ *          ],
+ *          moraydelink: [
+ *              '{region}_{shard}_moraydelink.sh',
+ *              ...
+ *          ]
+ *      }
+ * and `generated` is a boolean indicating if the delink scripts were newly
+ * generated in this call.
+ *
+ * If delink scripts already exist under "/var/db/snaplink-cleanup/delink/"
+ * then, this just returns the listing of them and ensures the full expected
+ * set is there.
+ */
+Migrator.prototype.ensureDelinkScripts = function ensureDelinkScripts(
+    opts,
+    cb
+) {
+    assert.string(opts.dbDir, 'opts.dbDir');
+    assert.arrayOfObject(opts.indexShards, 'opts.indexShards');
+
+    let delinkDir = path.join(opts.dbDir, 'delink');
+    let filenames;
+    let log = this.log;
+    let metadataCmdsFromShard = {};
+    let numZeroByteObjs = 0;
+    let scripts = {
+        stordelink: [],
+        moraydelink: []
+    };
+    let shardHosts;
+    let storCmdsFromStorId = {};
+    let storIds;
+
+    // mkdir -p $delinkDir
```

Comment:
nit: this isn't mkdir -p on the version of node that ships with sdc-manta, but I think we're ok because we've made sure that dbDir exists elsewhere. Perhaps just mention that here?

Reply:
Thanks. I just added creation of that dir to the mkDbDir step earlier that creates the DBDIR structure. See commit 2830340.